### PR TITLE
Feature: Union schema compatibility

### LINKED ISSUE #34: [Feature] Add ability to union multiple Google Ads connectors together
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Describe the Feature

I noticed that the Facebook Pages Source dbt package provides an option to union together multiple instances of Facebook Pages connectors for Fivetran: https://hub.getdbt.com/fivetran/facebook_pages_source/latest/. That option isn't available for the Google Ads Source package (https://hub.getdbt.com/fivetran/google_ads_source/latest/), but I am interested in having that included so I can utilize it.

This would likely use the existing `union_data` function in dbt_fivetran_utils package (https://github.com/fivetran/dbt_fivetran_utils/tree/v0.4.2/#union_data-source) similar to how it was done in Facebook Pages.

### Describe alternatives you've considered

We currently work around this limitation with two main changes:

1. We created models to union the source data together and push it to some central views
2. We override the variables that define the source input to this package to change them to use the `ref()` function (a bit hacky), which preserves the ability to have fully integrated scheduling in Fivetran. Here is an example of how the overrides were done in `dbt_project.yml`:
```
  google_ads_source:
    account_history: "{{ ref('stg_google_ads_combined__account_history') }}"
    account_stats: "{{ ref('stg_google_ads_combined__account_stats') }}"
    ad_group_criterion_history: "{{ ref('stg_google_ads_combined__ad_group_criterion_history') }}"
    ad_group_history: "{{ ref('stg_google_ads_combined__ad_group_history') }}"
    ad_group_stats: "{{ ref('stg_google_ads_combined__ad_group_stats') }}"
    ad_history: "{{ ref('stg_google_ads_combined__ad_history') }}"
    ad_stats: "{{ ref('stg_google_ads_combined__ad_stats') }}"
    campaign_history: "{{ ref('stg_google_ads_combined__campaign_history') }}"
    campaign_stats: "{{ ref('stg_google_ads_combined__campaign_stats') }}"
    geo_target: "{{ ref('stg_google_ads_combined__geo_target') }}"
    keyword_stats: "{{ ref('stg_google_ads_combined__keyword_stats') }}"
    label: "{{ ref('stg_google_ads_combined__label') }}"
```

### Are you interested in contributing this feature?

- [ ] Yes.
- [ ] Yes, but I will need assistance and will schedule time during your [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance.
- [X] No.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
Issue: #34 

Confirm the following files were correctly updated automatically:
- [x] CHANGELOG
- [x] README
- [x] stg_google_ads.yml, src_google_ads.yml, google_ads.yml (depending if source or transform)
- [x] docs.md
- [ ] ~joins~
- [x] window functions (partition by)
- [x] 'source_relation' column added in staging and transform models
- [x] tests

Manual updates:
- [x] Add `source_relation` to downstream models if necessary
- [x] Finish any incomplete/incorrect joins/partitions
- [x] Update tests to include `source_relation` in unique-combination-of-cols if necessary
    - May need to remove some uniqueness tests in favor of the combo test
    
Validation:
- [x] `dbt run` locally passes
- [x] `dbt test` locally passes
- See internal ticket for the validation that tests multiple sources

### CODE REVIEW

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md]: Actually, can we remove the README "hide" sections in these updates. This way we are able to link directly to the section in the README where the users needs to understand how to configure the union feature.

Otherwise, I worry this cool feature is hidden from people actually using it.

File: packages.yml
Comment (fivetran-joemarkiewicz) [inline packages.yml]: Reminder to update before merge

**Review by fivetran-joemarkiewicz — APPROVED** [review]
These updates look great! I have just a small request to update the CHANGELOG and README. Once those are updated, this will be good to go after the remaining package updates are approved.

File: CHANGELOG.md
Comment (fivetran-catfritz) [reply CHANGELOG.md]: Updated.

**Comment by fivetran-catfritz** [comment]
Merging for release on 12-Oct-2023.

**Pull request merged**
